operator/podinfo/podinfo_controller.go

```diff
@@ -0,0 +1,143 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package podinfo
+
+import (
+	"context"
+
+	ciliumiov1alpha1 "github.com/cilium/tetragon/pkg/k8s/apis/cilium.io/v1alpha1"
+	"github.com/google/go-cmp/cmp"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// Reconciler reconciles a PodInfo object
+type Reconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
```

Comment:
I don't see it being used, is it needed?

Reply:
yes, while setting up the controller, the client and scheme is being assigned. In the `cmd/serve/serve.go`

```Go
if err = (&podinfo.Reconciler{
	Client: mgr.GetClient(),
	Scheme: mgr.GetScheme(),
}).SetupWithManager(mgr); err != nil {
	setupLog.Error(err, "unable to create controller", "controller", "podinfo")
	os.Exit(1)
```

Here: https://github.com/cilium/tetragon/pull/1410/files#diff-a4c1a3f9a379da52152a1e2f245c40c5c541b3af8243cf61a66554a15f4d7d52R68-R74

Reply:
I see, I missed it earlier, thanks. But I still don't see this value read anywhere, am I missing something?

Reply:
yeah i think we just copied this from somewhere, and i don't know how or if it actually gets used. @prateek041 do you wanna try removing the `Scheme` field and see what happens? 🧑‍🔬

Reply:
most likely we don't need it because we don't call these functions:

- https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/controller/controllerutil#SetControllerReference
- https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/controller/controllerutil#SetOwnerReference

Reply:
sure ! let me take a look at this.

Reply:
I removed the scheme, the tests still pass. I guess we did not needed it 😄